diff --git a/cmake/f3dPlugin.cmake b/cmake/f3dPlugin.cmake
index 96d3d3b..bbfd32e 100644
--- a/cmake/f3dPlugin.cmake
+++ b/cmake/f3dPlugin.cmake
@@ -292,6 +292,7 @@ macro(f3d_plugin_build)
       MODULES ${modules}
       ENABLE_WRAPPING OFF
       INSTALL_HEADERS OFF
+      INSTALL_EXPORT f3dLibraryTargets
       PACKAGE "f3d_${F3D_PLUGIN_NAME}_vtkext_private")
 
     foreach (module IN LISTS modules)
@@ -363,9 +364,9 @@ macro(f3d_plugin_build)
     ${F3D_PLUGIN_VTK_MODULES}
     ${modules})
 
-  if(NOT F3D_PLUGIN_IS_STATIC)
+  if(1)
     install(TARGETS f3d-plugin-${F3D_PLUGIN_NAME}
-      EXPORT f3dTargets
+      EXPORT f3dLibraryTargets
       ARCHIVE DESTINATION ${_f3d_plugins_install_dir} COMPONENT plugin
       LIBRARY DESTINATION ${_f3d_plugins_install_dir} COMPONENT plugin)
   endif()
diff --git a/library/CMakeLists.txt b/library/CMakeLists.txt
index 287eb58..47c5086 100644
--- a/library/CMakeLists.txt
+++ b/library/CMakeLists.txt
@@ -255,7 +255,7 @@ install(
   EXCLUDE_FROM_ALL)
 
 ## Install the rest of the libraries and SDK parts
-if(BUILD_SHARED_LIBS)
+if(1)
 
   # Install the libf3d
   install(TARGETS libf3d
diff --git a/plugins/native/module/CMakeLists.txt b/plugins/native/module/CMakeLists.txt
index 55d822f..217eb61 100644
--- a/plugins/native/module/CMakeLists.txt
+++ b/plugins/native/module/CMakeLists.txt
@@ -7,7 +7,6 @@ set(classes
   )
 
 vtk_module_add_module(f3d::vtkextNative
-  NO_INSTALL
   FORCE_STATIC
   CLASSES ${classes})
 
diff --git a/vtkext/private/CMakeLists.txt b/vtkext/private/CMakeLists.txt
index ae748f0..93b8268 100644
--- a/vtkext/private/CMakeLists.txt
+++ b/vtkext/private/CMakeLists.txt
@@ -10,6 +10,7 @@ vtk_module_build(
   MODULES ${modules}
   ENABLE_WRAPPING OFF
   INSTALL_HEADERS OFF
+  INSTALL_EXPORT f3dLibraryTargets
   PACKAGE "f3d_vtkext_private")
 
 foreach (module IN LISTS modules)
diff --git a/vtkext/private/module/CMakeLists.txt b/vtkext/private/module/CMakeLists.txt
index 55484dc..bfb6984 100644
--- a/vtkext/private/module/CMakeLists.txt
+++ b/vtkext/private/module/CMakeLists.txt
@@ -145,7 +145,6 @@ if(F3D_MODULE_UI AND NOT F3D_USE_EXTERNAL_IMGUI)
 endif()
 
 vtk_module_add_module(f3d::vtkextPrivate
-  NO_INSTALL
   FORCE_STATIC
   CLASSES ${classes}
   SOURCES ${sources}
diff --git a/vtkext/public/CMakeLists.txt b/vtkext/public/CMakeLists.txt
index 92f1946..267ab80 100644
--- a/vtkext/public/CMakeLists.txt
+++ b/vtkext/public/CMakeLists.txt
@@ -18,7 +18,7 @@ set(headers_component "")
 set(f3d_vtk_no_install "")
 set(f3d_vtk_force_static "")
 get_target_property(f3d_vtk_target_type VTK::CommonCore TYPE)
-if(BUILD_SHARED_LIBS AND NOT f3d_vtk_target_type STREQUAL STATIC_LIBRARY)
+if(1)
   # The headers and all "dev" part are installed as part of the plugin SDK
   # The library itself is installed as part of the library
   set(export_name "f3d_vtkext")
