diff --git a/src/osgPlugins/curl/CMakeLists.txt b/src/osgPlugins/curl/CMakeLists.txt
index 667631f..7a553f9 100644
--- a/src/osgPlugins/curl/CMakeLists.txt
+++ b/src/osgPlugins/curl/CMakeLists.txt
@@ -23,17 +23,17 @@ SET(TARGET_H
 
 IF(ZLIB_FOUND)   
     SET(TARGET_LIBRARIES_VARS
-        CURL_LIBRARY
+        CURL_LIBRARIES
         ZLIB_LIBRARIES)
 ELSE()
     SET(TARGET_LIBRARIES_VARS
-        CURL_LIBRARY)
+        CURL_LIBRARIES)
 ENDIF()
 
 IF(WIN32 OR MINGW)
     IF(CURL_IS_STATIC)
         ADD_DEFINITIONS(-DCURL_STATICLIB)
-        SET(TARGET_EXTERNAL_LIBRARIES ${TARGET_EXTERNAL_LIBRARIES} ws2_32 winmm wldap32)
+        SET(TARGET_EXTERNAL_LIBRARIES ${TARGET_EXTERNAL_LIBRARIES} ws2_32 winmm wldap32 crypt32)
     ENDIF()
 ENDIF()
 
